marathon-cli.py

```diff
@@ -0,0 +1,331 @@
+    MARATHON_FORCE_DEPLOY:      Use the Force, if necessary (i.e. when a
+                                deployment is failing.)
+    MARATHON_FRAMEWORK_NAME:    The name of the framework (usually 'marathon')
+    MARATHON_APP:               The JSON formatted app definition.
+    MARATHON_RETRIES:           The number of task failures to tolerate.
+    MESOS_AGENT_MAP:            This is used when Mesos is behind a proxy.  The
+                                API will return the Mesos Agent IP address,
+                                but that may need to be mapped to a URL.  The
+                                map is defined like:
+                                10.0.1.34|https://mesos.test.dev,...
+    MESOS_MASTER_URLS:          One or more urls for Mesos communication,
+                                separated by commas.
+    """
+
+    marathon_urls = os.getenv("MARATHON_URLS", "http://localhost:8080").split(',')
+    marathon_app_id = os.getenv("MARATHON_APP_ID", "test-app")
+    marathon_user = os.getenv("MARATHON_USER", None)
+    marathon_password = os.getenv("MARATHON_PASSWORD", None)
+    marathon_force = True if os.getenv("MARATHON_FORCE_DEPLOY", "false") == "true" else False
+    marathon_framework_name = os.getenv("MARATHON_FRAMEWORK_NAME", "marathon")
+    marathon_retries = os.getenv("MARATHON_RETRIES", 3)
+    log_level = os.getenv("MARATHON_LOGLEVEL", 'info')
+    marathon_app = os.getenv("MARATHON_APP","""
+        {
+            "id": "/test-app",
+            "cmd": "mv *.war apache-tomcat-*/webapps && cd apache-tomcat-* && sed \\"s/8080/$PORT/g\\" < ./conf/server.xml > ./conf/server-mesos.xml && sleep 15 && ./bin/catalina.sh run -config ./conf/server-mesos.xml",
+            "cpus": 1,
+            "mem": 128,
+            "disk": 0,
+            "instances": 1,
+            "user": "mesagent",
+            "uris": [
+                "http://mirrors.gigenet.com/apache/tomcat/tomcat-7/v7.0.73/bin/apache-tomcat-7.0.73.tar.gz",
+                "https://storage.googleapis.com/google-code-archive-downloads/v2/code.google.com/gwt-examples/Calendar.war"
+            ],
+            "healthChecks": [
+                {
+                  "protocol": "HTTP",
+                  "path": "/",
+                  "portIndex": 0,
+                  "gracePeriodSeconds": 300,
+                  "intervalSeconds": 60,
+                  "timeoutSeconds": 20,
+                  "maxConsecutiveFailures": 3
+                }
+            ]
+        }
+    """)
+    mesos_agent_map_string = os.getenv("MESOS_AGENT_MAP", None)
+    mesos_master_urls = os.getenv("MESOS_MASTER_URLS", "http://localhost:5050").split(',')
+
+    exit_code = 0
+    auth = None
+    if marathon_user and marathon_password:
+        auth = (marathon_user, marathon_password)
+
+    ### Setup Logging
+    logging.basicConfig(format="%(levelname)-8s %(message)s", level=getattr(logging, log_level.upper()))
```

Comment:
It's probably worth quickly splitting all of these chunks of functionality (indicated by `###` comments)  into their own methods so you are more explicit about any state that they're changing.